drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,572 @@
+/* sensor_apds9960.c - driver for APDS9960 ALS/RGB/gesture/proximity sensor */
+
+/*
+
+  Copyright (c) 2017 Intel Corporation
+
+  SPDX-License-Identifier: Apache-2.0
+
+*/
+
+#include "apds9960.h"
+
+uint8_t apds9960_getmode(struct device *dev)
+{
+    uint8_t enable_value;
+    struct apds9960_data *data = dev->driver_data;
+
+    /* Read current ENABLE register */
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS, 0x80 ,&enable_value) < 0){
+                SYS_LOG_DBG(" Failed");
+		return 0xFF;
+        }
+
+    return enable_value;
+}
+
+bool apds9960_setmode(struct device *dev, uint8_t mode, uint8_t enable)
+{
+	uint8_t reg_val;
+	struct apds9960_data *data = dev->driver_data;
+
+	/* Read current ENABLE register */
+	reg_val = apds9960_getmode(dev);
+
+	if(reg_val == 0xFF) {
+		return false;
+	}
+
+	/* Change bit(s) in ENABLE register */
+	enable = enable & 0x01;
+	if( mode >= 0 && mode <= 6 ) {
+		if (enable) {
+		reg_val |= (1 << mode);
+		} else {
+		reg_val &= ~(1 << mode);
+		}
+
+	} else if( mode == APDS9960_ALL ) {
+		if (enable) {
+		reg_val = 0x7F;
+		} else {
+		reg_val = 0x00;
+		}
+	}
+
+	/* Write value back to ENABLE register */
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS, 0x80 , reg_val) < 0) {
+                SYS_LOG_DBG(" Failed");
+		return false;
+        }
+
+    return true;
+}
+
+bool apds9960_enablepower(struct device *dev)
+{
+	if (!apds9960_setmode(dev, APDS9960_POWER, 1)) {
+		return false;
+	}
+
+	return false;
+}
+
+static int apds9960_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct apds9960_data *data = dev->driver_data;
+	uint8_t cmsb, clsb, rmsb, rlsb, blsb, bmsb, glsb, gmsb;
+	uint8_t pdata;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* Read lower byte following by MSB  Ref: Datasheet : RGBC register */
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_CDATAL_REG, &clsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_CDATAH_REG, &cmsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_RDATAL_REG, &rlsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_RDATAH_REG, &rmsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_GDATAL_REG, &glsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_GDATAH_REG, &gmsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_BDATAL_REG, &blsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_BDATAH_REG, &bmsb) < 0) {
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
+			APDS9960_PDATA_REG, &pdata) < 0) {
+		return -EIO;
+	}
```

Comment:
These registers are all contiguous so you should be able to replace all the i2c_reg_read_byte calls with one i2c_burst_read. Unless this sensor doesn't support address auto-increment?